0019_ldap_deprecated.dpatch by  dann frazier <dannf@debian.org>

The function ldap_get_values, used in saslauthd/lak.c, is deprecated.
Therefore, its prototype is not included by default when compiling
against the ldap.h headers. As a result, the compiler cannot know the
return type of the function at compile time, and will implicitly
convert to a pointer. This has implications on 64-bit systems.
This patch sets the deprecation mode on, so that the function prototype
gets included when compiling.
(Description by Fabian Fagerholm <fabbe@debian.org>)

diff -urNad trunk~/saslauthd/lak.c trunk/saslauthd/lak.c
--- trunk~/saslauthd/lak.c	2006-06-04 12:26:20.000000000 +0300
+++ trunk/saslauthd/lak.c	2008-02-15 14:32:11.000000000 +0200
@@ -55,6 +55,7 @@
 #include <openssl/des.h>
 #endif
 
+#define LDAP_DEPRECATED 1
 #include <ldap.h>
 #include <lber.h>
 #include <sasl.h>
